### ISSUE TITLE
React Flow doesn't work in Shadow DOM

### ISSUE THREAD

**Issue by KislyakovDS** [wbkd/react-flow#2311]
### Describe the Bug

I need to use React Flow inside Shadow DOM, but I ran into several errors:

1) Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.

![image](https://user-images.githubusercontent.com/18344847/180201855-8b9630b0-a1ca-45d7-baf0-f38f401ee265.png)

I made a small local fix in node_modules to check new behavior:
replaced this `document.querySelector('.react-flow__viewport')`
whith this `document.querySelector('#shadow').shadowRoot.querySelector('.react-flow__viewport')` - `'#shadow'` is my element with Shadow DOM.

And I got the following bug

2) There are broken view for nodes and edges

![image](https://user-images.githubusercontent.com/18344847/180203235-88d2928c-d938-4513-a67a-9b8dbf59d756.png)

### Your Example Website or App

https://github.com/KislyakovDS/react-flow-with-shadow-dom

### Steps to Reproduce the Bug or Issue

1. Go to https://react-flow-with-shadow-dom.vercel.app/
2. Open browser console to see error

### Expected behavior

As a user, I expected behavior from this example https://reactflow.dev/docs/examples/overview/

### Platform

- OS: [Windows]
- Browser: [Chrome]
- Version: [103.0.5060.114]

**Comment by moklick** [comment]
Thanks for the report and the repo! We will try to fix it!

**Comment by moklick** [comment]
Could you try if the issue still occurs when you use the `next` release: `npm install react-flow-renderer@next`?

Did you add the [default styles](/src/style.css) to the shadow root?

**Comment by KislyakovDS** [comment]
@moklick 
The first error is gone - ok.

About the second error: I added only this style `react-flow-renderer/dist/style.css` to the shadow root. Now I have added another style `react-flow-renderer/dist/theme-default.css` and seems all right.

I have pushed new version to my repo and deploy it.

![image](https://user-images.githubusercontent.com/18344847/180763053-faaf3dc2-98f3-411f-82e8-df2940c894b7.png)
